From 9d67479e94e843d95bceb5bf596b4479f2035fe8 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 17 Jul 2015 11:43:31 +0800
Subject: [PATCH 0215/5242] MLK-11258 tty: serial: imx: disable overrun
 interrupt during uart port shutdown

commit  9433a2d8b264f64d7d4ac5d4145bb4e1797c55b3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Also disable rx fifo overrun interrupt during uart port shutdown.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: 39eb703f6c9f9359723f3fa22e798b1d21b44c67)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index e313de3..e9d2df3 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -465,10 +465,11 @@ static void imx_uart_stop_tx(struct uart_port *port)
 static void imx_uart_stop_rx(struct uart_port *port)
 {
 	struct imx_port *sport = (struct imx_port *)port;
-	u32 ucr1, ucr2;
+	u32 ucr1, ucr2, ucr4;
 
 	ucr1 = imx_uart_readl(sport, UCR1);
 	ucr2 = imx_uart_readl(sport, UCR2);
+	ucr4 = imx_uart_readl(sport, UCR4);
 
 	if (sport->dma_is_enabled) {
 		ucr1 &= ~(UCR1_RXDMAEN | UCR1_ATDMAEN);
@@ -480,6 +481,9 @@ static void imx_uart_stop_rx(struct uart_port *port)
 
 	ucr2 &= ~UCR2_RXEN;
 	imx_uart_writel(sport, ucr2, UCR2);
+
+	ucr4 &= ~UCR4_OREN;
+	imx_uart_writel(sport, ucr4, UCR4);
 }
 
 /* called with port.lock taken and irqs off */
-- 
1.7.9.5

